samplePDF/FarDetectorCoreInfoStruct.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+/// @brief constructors are same for all three so put in here
+struct FarDetectorCoreInfo {
+  int nutype; // 2 = numu/signue | -2 = numub | 1 = nue | -1 = nueb           
+  int oscnutype;    
+  int nupdg;
+  int nupdgUnosc;
+  bool signal; // true if signue                                              
+  int nEvents; // how many MC events are there                              
+  std::string flavourName;
+
+  std::vector<int*> Target; // target the interaction was on
+
+  int SampleDetID;
+
+  //THe x_var and y_vars that you're binning in
+  std::vector<const double*> x_var;
+  std::vector<const double*> y_var;
+  std::vector<const double*> rw_etru;
+  std::vector<const double*> rw_truecz;
+
+  /// xsec bins
+  std::list< int > *xsec_norms_bins;
+
+  /// DB Speedup bits
+  double Unity;
+  float Unity_F;
+  int Unity_Int;
+  double dummy_value = -999;
+
+  std::vector<int> nxsec_norm_pointers;
+  std::vector<std::vector<const double*>> xsec_norm_pointers;
+
+  std::vector<int> nxsec_spline_pointers;
+  std::vector<std::vector<const double*>> xsec_spline_pointers;
+
+  std::vector<int> ntotal_weight_pointers;
+  std::vector<std::vector<const double*>> total_weight_pointers;
+  std::vector<double> total_w;
+
+  std::vector<int> XBin;
+  std::vector<int> YBin;
+  std::vector<int> NomXBin;
+  std::vector<int> NomYBin;
+
+  bool *isNC;
+
+  // histo pdf bins
```

Comment:
Why is it raw pointer while everyhitng else is vector?

Reply:
Read about std::vector<bool>, it doesn't behave like other std::vector objects, or just try printing the address of the first three entries of auto myvecbool = std::vector<bool>{true,true,false};

Reply:
I feel dumb for not knowing this...

In this case there should be a destructor?